### PR TITLE
Handle remote data source

### PR DESCRIPTION
**Pull request by nvquangth**
Function: add remote data source and repository
Redmine: https://edu-redmine.framgia.vn/issues/1250

### CODE REVIEW

**Comment by nvquangth** [comment]
ready

File: app/src/main/java/com/quangnv/moviedb/data/repository/GenreRepository.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        return sInstance;
+    }
+
+    @Override
+    public void saveGenre(Genre genre) {
+
+    }
+
+    @Override
+    public void deleteGenres(Genre genre) {
```
Comment (tuannguyen38) [inline app/src/main/java/com/quangnv/moviedb/data/repository/GenreRepository.java]: rename to deleteGenre

File: app/src/main/java/com/quangnv/moviedb/data/repository/MovieRepository.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    }
+
+    @Override
+    public void deleteMovie(Movie movie) {
+
+    }
+
+    @Override
+    public void deleteAllMovie() {
```
Comment (tuannguyen38) [inline app/src/main/java/com/quangnv/moviedb/data/repository/MovieRepository.java]: Movies

File: app/src/main/java/com/quangnv/moviedb/util/Constant.java [hunk 3]
```diff
@@ -7,4 +7,5 @@
 public final class Constant {
```
Comment (tuannguyen38) [inline app/src/main/java/com/quangnv/moviedb/util/Constant.java]: Constants

**Comment by nvquangth** [comment]
ready

**Pull request merged**